site/en/blog/headless-chrome/index.md

```diff
@@ -0,0 +1,608 @@
+updated: 2018-02-23
+
+# Optional
+# How to add a new tag
+# https://developer.chrome.com/docs/handbook/how-to/add-a-tag/
+tags:
+  - testing
+  - headless
+  - chrome59
+
+
+
+---
+
+
+<style>
+figure {
+  text-align: center;
+}
+</style>
+
+## TL;DR
+
+[Headless Chrome](https://chromium.googlesource.com/chromium/src/+/lkgr/headless/README.md)
+is shipping in Chrome 59. It's a way to run the Chrome browser in a headless environment.
+Essentially, running
+Chrome without chrome! It brings **all modern web platform features** provided
+by Chromium and the Blink rendering engine to the command line.
+
+Why is that useful?
+
+A headless browser is a great tool for automated testing and server environments where you
+don't need a visible UI shell. For example, you may want to run some tests against
+a real web page, create a PDF of it, or just inspect how the browser renders an URL.
+
+{% Aside %}
+Note: Headless mode has been available on Mac and Linux since **Chrome 59**.
+[Windows support](https://bugs.chromium.org/p/chromium/issues/detail?id=686608)
+came in Chrome 60.
+{% endAside %}
+
+## Starting Headless (CLI)
+
+The easiest way to get started with headless mode is to open the Chrome binary
+from the command line. If you've got Chrome 59+ installed, start Chrome with the `--headless` flag:
+
+```shell
+    chrome \
+      --headless \                   # Runs Chrome in headless mode.
+      --disable-gpu \                # Temporarily needed if running on Windows.
+      --remote-debugging-port=9222 \
+      https://www.chromestatus.com   # URL to open. Defaults to about:blank.
+```
+
+{% Aside %}
+Note: Right now, you'll also want to include the `--disable-gpu` flag if you're running
+on Windows. See [crbug.com/737678](https://bugs.chromium.org/p/chromium/issues/detail?id=737678).
+{% endAside %}
+
+`chrome` should point to your installation of Chrome. The exact location will
+vary from platform to platform. Since I'm on Mac, I created convenient aliases
+for each version of Chrome that I have installed.
+
+If you're on the stable channel of Chrome and cannot get the Beta, I recommend
+using `chrome-canary`:
+
+```shell
+    alias chrome="/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome"
+    alias chrome-canary="/Applications/Google\ Chrome\ Canary.app/Contents/MacOS/Google\ Chrome\ Canary"
+    alias chromium="/Applications/Chromium.app/Contents/MacOS/Chromium"
+```
+
+Download Chrome Canary [here](https://www.google.com/chrome/browser/canary.html).
+
+## Command line features
+
+In some cases, you may not need to [programmatically script](#node) Headless Chrome.
+There are some [useful command line flags](https://cs.chromium.org/chromium/src/headless/app/headless_shell_switches.cc)
+to perform common tasks.
+
+### Printing the DOM
+
+The `--dump-dom` flag prints `document.body.innerHTML` to stdout:
+
+    chrome --headless --disable-gpu --dump-dom https://www.chromestatus.com/
```

Comment:
Sorry! Completely missed that block of code. 

Ok cool. Didn't know that without the language label it would be ok. I'll try to always assign it, but it's useful to have that in mind. Thanks!

Reply:
If you can't figure it out, don't worry. I can always make a suggestion.